src/actions/who-am-i/index.ts

```diff
@@ -1,23 +1,25 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
-import * as core from '@actions/core';
-import { AuthKind, AuthHandler, DefaultRunnerFactory, RunnerFactory } from '../../lib';
+import * as core from "@actions/core";
+import { AuthKind, AuthHandler } from "../../lib";
+import createActionsPacRunner from "../../lib/createActionsPacRunner";
+import createCliWrapperPacAuthenticator from "../../lib/auth/createCliWrapperPacAuthenticator";
+import { PacRunner } from "@microsoft/powerplatform-cli-wrapper";
 
 (async () => {
     if (process.env.GITHUB_ACTIONS) {
-        await main(DefaultRunnerFactory);
+        await main(createActionsPacRunner());
     }
 })();
 
-export async function main(factory: RunnerFactory): Promise<void> {
+export async function main(pac: PacRunner): Promise<void> {
     try {
-        core.startGroup('who-am-i');
+        core.startGroup("who-am-i");
 
-        const pac = factory.getRunner('pac', process.cwd());
-        await new AuthHandler(pac).authenticate(AuthKind.CDS);
+        const authenticator = createCliWrapperPacAuthenticator(pac);
+        await new AuthHandler(authenticator).authenticate(AuthKind.CDS);
 
-        const whoArgs = ['org', 'who'];
-        await pac.run(whoArgs);
+        await pac.whoAmI();
```

Comment:
thanks, looks much more cleaner with the npm package!
while I would expect that who-am-i for now will still throw on linux (since the pp-cli-wrapper npm pkg has not been updated with Crash's linux changes), I'm a tad worried that it doesn't fail the linux PR build leg... but it should!
Note I'm fine for this PR if the flow would skip who-am-i for linux, once the action is updated to fail due to not-supported-yet

(node:2013) UnhandledPromiseRejectionWarning: Error: Unsupported Action runner os: 'linux'; for the time being, only Windows runners are supported (cross-platform support work is in progress)
    at Object.restrictPlatformToWindows [as default] (/home/runner/work/powerplatform-actions/powerplatform-actions/dist/actions/who-am-i/index.js:571:15)
    at Object.createCommandRunner (/home/runner/work/powerplatform-actions/powerplatform-actions/dist/actions/who-am-i/index.js:415:40)
    at Object.createPacRunner (/home/runner/work/powerplatform-actions/powerplatform-actions/dist/actions/who-am-i/index.js:521:40)
    at Object.createActionsPacRunner [as default] (/home/runner/work/powerplatform-actions/powerplatform-actions/dist/actions/who-am-i/index.js:974:40)
    at /home/runner/work/powerplatform-actions/powerplatform-actions/dist/actions/who-am-i/index.js:625:52
    at Generator.next (<anonymous>)
    at /home/runner/work/powerplatform-actions/powerplatform-actions/dist/actions/who-am-i/index.js:612:71
    at new Promise (<anonymous>)
    at module.exports.__webpack_modules__.274.__awaiter (/home/runner/work/powerplatform-actions/powerplatform-actions/dist/actions/who-am-i/index.js:608:12)
    at /home/runner/work/powerplatform-actions/powerplatform-actions/dist/actions/who-am-i/index.js:623:8
(node:2013) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)

Reply:
Oh wow, good catch! I'm going to get it to fail first and then have the linux one skip for now (to update in another PR).

Reply:
perfect, thanks!